### PR TITLE
Fix #308

### LINKED ISSUE #308: $ and ` considered numbers
The characters `$` and `` ` ``, by themselves, are considered numbers by rgbasm. This seems to act as a degenerate case of a hexadecimal number or a GFX number where the number has no digits, and thus has the value zero. (This doesn't happen for `%`, which I assume is special-cased since it's also an operator.)

For instance, the following code:
```
  ld d, $
  ld hl, `
```

...will load 0 into `d` and `hl`.

This doesn't seem desirable at all.

### PR DESCRIPTION
**Pull request by dbrotz**
Fixes #308.

### CODE REVIEW

**Comment by AntonioND** [comment]
Same as the other PR: The code looks fine, but the commit message should be descriptive and not refer to a GitHub issue. If the project were to move away from here the message wouldn't be useful anymore.

File: src/asm/globlex.c [hunk 1]
```diff
@@ -100,7 +100,10 @@ static int32_t ascii2bin(char *s)
 		break;
 	}
 
-	if (radix == 4) {
+	if (radix != 10 && *s == '\0') {
```
Comment (AntonioND) [inline src/asm/globlex.c]: Do we really need the "radix != 10"? As far as I see, the only way to have radix = 10 is to have a decimal number (with no prefix), so the only way to get an empty number with radix = 10 is a serious parser problem that shouldn't happen (passing an empty string to this function).

Can't you remove it?

**Comment by AntonioND** [comment]
Also, this changes the output of some tests. You need to regenerate the references (run ``tests/*/update-refs.sh``).

File: src/asm/globlex.c (hunk 1 above)
Comment (dbrotz) [reply src/asm/globlex.c]: I agree. It's not necessary.

**Pull request merged**